src/Tickets/Commerce/Gateways/Stripe/Logger.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace TEC\Tickets\Commerce\Gateways\Stripe;
+
+/**
+ * Class Logger for Stripe errors
+ *
+ * @since TBD
+ *
+ * @package TEC\Tickets\Commerce\Gateways\Stripe;
+ */
+class Logger {
+
+	/**
+	 * Maximum number of error log entries to keep
+	 *
+	 * @since TBD
+	 *
+	 * @var int
+	 */
+	const MAX_LOG_SIZE = 200;
+
+	/**
+	 * How long should we keep logs entries for
+	 *
+	 * @since TBD
+	 *
+	 * @var int
+	 */
+	const LOG_EXPIRES_IN = DAY_IN_SECONDS;
+
+	/**
+	 * Option name for the error log option
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public static $stripe_error_log_option = 'tickets-commerce-stripe-error-log';
```

Comment:
Since this is used as a prefix make sure that is added on the variable name. Since transients are global lets go with `tec-tc-stripe-error-` to allow it to be shorter since transients can have a char limit.